fake/src/lib.rs

```diff
@@ -31,6 +140,24 @@ mod private {
     }
 }
 
+/// Generate fake values.
+///
+/// Generate default fake values with [`Faker`].
+/// Generate specific fake values with helpers in [`faker`].
+///
+/// This trait is implemented for any type that implements [`Dummy`]:
+/// [`Dummy`] should be implemented istead, and you get the [`Fake`]
+/// implementation for free.
+///
+/// # Examples
+///
+/// ```
+/// use fake::Fake;
+///
+/// assert_eq!(10.fake::<String>().len(), 10);
+/// let a: [[u8; 2]; 3] = (1..10).fake();
+/// let b: Option<Option<usize>> = (1..10).fake();
+/// ```
 pub trait Fake: Sized {
```

Comment:
I don't quite know how to explain the difference between `Fake` and `Dummy` (see the first line).

Reply:
`Dummy` is similar to `From` trait, while `Fake` is similar to `Into` trait.
And the lib provided blanket implementation for `Fake` if type implemented `Dummy`

The calling style is different, below both generate fake string with length between 1 to 8
```rust
let a = String::dummy(1..8);
let b = (1..8).fake();
```

Reply:
But one cannot implement `Fake` right?

Reply:
I think so, as there has `impl<T> Fake for T {}` in the crate. User only need to implement `Dummy`

Reply:
But at first look `Fake` is related to `Dummy` does not seemed straightforward, there is also `Faker` which may be why I am confused.

Reply:
We can generate fake value in target type `T` from configuration type `U`, if `T: Dummy<U>`. `Faker` is one of configuration type without arguments and provide default generation logic, many type implemented `Dummy<Faker>`, so that we can use like below.
```rust
let a: u8 = Faker.fake(); // Default u8 generation logic, or u8::dummy(Faker)
let b: f32 = Faker.fake(); // Default f32 generation logic, or f32::dummy(Faker)
```
However, in order to control the generation logic for given target type T, we need different configuration type. e.g.
```rust
let a: u8 = Faker.fake(); // Default u8 generation logic
let b: u8 = (1..8).fake(); // generate random u8 in range(1, 8)

// u8 can generated from Faker or Range
let a = u8::dummy(Faker);
let b = u8::dummy(1..8);

// to have your own u8 fake value generation logic, you create new configuration type
struct EvenU8;
impl Dummy<EvenU8> for u8 {
  // generate even number
  ...
}

let even_num: u8 = EvenU8.fake();
let even_num = u8::dummy(EvenU8); 
```